library/test/src/term/terminfo/mod.rs

```diff
@@ -119,13 +130,22 @@ pub(crate) struct TerminfoTerminal<T> {
 impl<T: Write + Send> Terminal for TerminfoTerminal<T> {
     fn fg(&mut self, color: color::Color) -> io::Result<bool> {
         let color = self.dim_if_necessary(color);
+        if cfg!(miri) && color < 8 {
```

Comment:
It's not obvious to me what happens if `color = 9`. The logic for `num_colors` isn't totally straightforward, but I think in Miri we end up with `num_colors = 0`, right? Could we explain this in a comment?

Reply:
Yeah `num_colors` is always 0 in Miri. I added a comment.